drivers/hmis/app/graphql/types/application/user_dashboard.rb

```diff
@@ -0,0 +1,70 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Types
+  class Application::UserDashboard < Types::BaseObject
+    # Type for resolving data needed on the HMIS user dashboard.
+    # Underlying object is Hmis::User, who should always be the currently logged-in user.
+    description 'Resolves everything that is needed on the user dashboard'
+
+    field :id, ID, null: false
+    field :user_dashboard_config, Types::Application::UserDashboardConfig, null: false
+    field :staff_assignments, HmisSchema::StaffAssignment.page_type, null: true
+    field :ce_referral_steps, HmisSchema::CeReferralStep.page_type, null: true
+
+    def user_dashboard_config
+      {
+        id: object.id,
+        show_staff_assignment: show_staff_assignment,
+        show_referrals: show_referrals,
+      }
+    end
+
+    def staff_assignments
+      object.staff_assignments.
+        viewable_by(current_user).
```

Comment:
I think its a little unintuitive that we need the viewable_by scope even though we're already scoped to the users assignments, maybe worth a comment:
```suggestion
        viewable_by(current_user). # Only resolve assignments where the user has access to view the household
```